src/app/modules/shared-components/components/path-suggestion/path-suggestion.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, Input, OnChanges, OnDestroy } from '@angular/core';
+import {
+  BreadcrumbsFromRoot,
+  GetBreadcrumbsFromRootsService
+} from '../../../item/http-services/get-breadcrumbs-from-roots.service';
+import { ReplaySubject, Subject, switchMap } from 'rxjs';
+import { mapToFetchState } from '../../../../shared/operators/state';
+import { map } from 'rxjs/operators';
+import { itemRoute } from '../../../../shared/routing/item-route';
+import { ItemRouter } from '../../../../shared/routing/item-router';
+
+@Component({
+  selector: 'alg-path-suggestion',
+  templateUrl: './path-suggestion.component.html',
+  styleUrls: [ './path-suggestion.component.scss' ],
+})
+export class PathSuggestionComponent implements OnDestroy, OnChanges {
+  @Input() itemId?: string;
+
+  private readonly itemId$ = new ReplaySubject<string>(1);
+  private readonly refresh$ = new Subject<void>();
+
+  state$ = this.itemId$.pipe(
+    switchMap(itemId => this.getBreadcrumbsFromRootsService.get(itemId).pipe(
+      map(group => {
+        if (group.length === 0) {
+          return undefined;
+        }
+
+        const allBreadcrumbs = group
+          .map(items => items.slice(0, items.length - 1))
+          .reduce((state, items) => [ ...state, ...items ], []);
+
+        if (allBreadcrumbs.length === 0) {
+          return [];
+        }
```

Comment:
I think these lines don't change anything. If I'm correct, the whole point of these lines is to return an empty array when all breadrcumbs are empty.
To me, this case is already handled on the return line: if after applying `.map(...).filter(...)` you had only empty breadcrumbs, it will also return an empty array.

Reply:
Base on conversation with Damien:

(empty path != no path). In case of empty path, you can display "This content is one of your root activity/skill"

There must be 2 condition:

1. If API returns empty (no path): You have no path to this content.
2. If API returns just 1 element, which is current and which is sliced (empty path): This content is one of your root activity/skill

Reply:
Fixed